packages/core/src/util/txInspector.ts

```diff
@@ -0,0 +1,145 @@
+import {
+  Address,
+  CertificateType,
+  Ed25519KeyHash,
+  Lovelace,
+  PoolId,
+  StakeAddressCertificate,
+  StakeDelegationCertificate,
+  TokenMap,
+  TxAlonzo,
+  TxOut,
+  Value
+} from '../Cardano';
+
+const isMyAddress =
+  (addresses: Address[]) =>
+  ({ address }: { address: Address }) =>
+    addresses.includes(address!);
+
+const isOutgoing = (tx: TxAlonzo, ownAddresses: Address[]): boolean => tx.body.inputs.some(isMyAddress(ownAddresses));
+
+const getTotalValue = (outputs: TxOut[]): Value => {
+  const result: Value = { coins: 0n };
+  const totalAssets: TokenMap = new Map();
+
+  for (const output of outputs) {
+    result.coins += output.value.coins;
+    if (output.value.assets) {
+      for (const [asset, amount] of output.value.assets) {
+        const total = totalAssets.get(asset) ?? 0n;
+        totalAssets.set(asset, total + amount);
+      }
+    }
+  }
+
+  if (totalAssets.size > 0) result.assets = totalAssets;
+  return result;
+};
```

Comment:
Could these methods be moved somewhere else? `isMyAddress` and `isOutgoing` were pretty much copied from `TransactionsTracker`

Reply:
`Address/util.ts`

Reply:
Please also add docblocks after hoisting

Reply:
Done in 19ae735f102d9a785ed2ab7a84f055380eb1ce77, ended up removing `getTotalValue` because I realized that `coalesceValueQuantities` exists